src/python/dxpy/utils/executable_unbuilder.py

```diff
@@ -115,85 +116,118 @@ def make_cluster_bootstrap_script_file(region, entry_point, code, suffix):
-                                              "folder": asset_record.describe().get("folder"),
-                                              "version": asset_record.describe(fields={"properties": True}
-                                                                               )["properties"]["version"]
-                                              }
-                        if dep.get("stages"):
-                            asset_json["stages"] = dep["stages"]
-                        asset_depends.append(asset_json)
-                        deps_to_remove.append(dep)
-                    except DXError:
-                        print("Describe failed on the assetDepends record object with ID - " +
-                              asset_record_id + "\n", file=sys.stderr)
-                        pass
-            else:
-                break
-    # Reversing the order of the asset_depends[] so that original order is maintained
-    asset_depends.reverse()
-    # resources/ directory
-    created_resources_directory = False
+    
+    # Get regions where the user's billTo are permitted
+    try:
+        bill_to = dxpy.api.user_describe(dxpy.whoami())['billTo']
+        permitted_regions = set(dxpy.DXHTTPRequest('/' + bill_to + '/describe', {}).get("permittedRegions"))
+    except DXError:
+        print("Failed to get permitted regions where {} can perform billable activities.\n".format(bill_to), file=sys.stderr)
+        sys.exit(1)    
+    
+    # when applet/app is built, the runSpec is initialized with fields "interpreter" and "bundledDependsByRegion"
+    # even when we don't have any bundledDepends in dxapp.json
+    enabled_regions = set(info["runSpec"]["bundledDependsByRegion"].keys())
+    if not enabled_regions.issubset(permitted_regions):
+        print("Region(s) {} are not among the permitted regions of {}. Resources from these regions will not be available.".format(
+            ", ".join(enabled_regions.difference(permitted_regions)), bill_to), file=sys.stderr )
+    # Update enabled regions
+    enabled_regions.intersection_update(permitted_regions)
+    
+    # Start downloading when not omitting resources
+    deps_downloaded = set()
     if not omit_resources:
-        for dep in info["runSpec"]["bundledDepends"]:
-            if dep in deps_to_remove:
-                continue
-            handler = get_handler(dep["id"])
-            if isinstance(handler, dxpy.DXFile):
+        download_completed = False
+        
+        # Check if at least one region is enabled 
+        if not enabled_regions:
+            raise DXError(
+                "Cannot download resources of the requested executable {} since it is not available in any of the billable regions. "
+                "You can use the --omit-resources flag to skip downloading the resources. ".format(info["name"]))
+        
+        print("Dependencies could be retrieved from region: {}. ".format(", ".join(enabled_regions)), file=sys.stderr)
```

Comment:
One additional comment. IMHO this message is not that useful since it prints what we know from the app spec. I'd remove it so that there is not too much stuff printed to the terminal